amm/drink-tests/src/router_tests.rs

```diff
@@ -283,7 +284,173 @@ fn test_fees(mut session: Session) {
+        .to_account_id()
+        .into();
+
+    // Add new pair to router cache
+    session
+        .execute(router.add_pair_to_cache(ice_wood_pair.into()))
+        .unwrap();
+
+    let amount = 1_000_000 * TOKEN;
+    psp22_utils::transfer(&mut session, ice.into(), ice_wood_pair.into(), amount, BOB).unwrap();
+    psp22_utils::transfer(&mut session, wood.into(), ice_wood_pair.into(), amount, BOB).unwrap();
+
+    let liquidity_minted = session
+        .execute(ice_wood_pair.mint(bob()))
+        .unwrap()
+        .result
+        .unwrap()
+        .unwrap();
+
+    let bob_lp_balance: u128 = psp22_utils::balance_of(&mut session, ice_wood_pair.into(), bob());
+    assert_eq!(liquidity_minted, bob_lp_balance);
+
+    let deadline = now + 10;
+
+    // Perform swap
+    let swap_amount = 10_000 * TOKEN;
+    psp22_utils::increase_allowance(&mut session, ice.into(), router.into(), swap_amount, BOB)
+        .unwrap();
+    let swap_res = session
+        .execute(router.swap_exact_tokens_for_tokens(
+            swap_amount,
+            0,
+            vec![ice.into(), wood.into()],
+            bob(),
+            deadline,
+        ))
+        .unwrap()
+        .result
+        .unwrap()
+        .unwrap();
+
+    // We cannot assert that as it depends on the pool's liquidity and swap amount. The price will move from the spot price.
+    // assert!(swap_res[1] >= min_amount_out);
+
+    // No fees distributed until the burn/mint transaction.
+    assert!(psp22_utils::balance_of(&mut session, ice_wood_pair.into(), charlie()) == 0);
+    // Burn some liquidity to trigger fee collection.
+    psp22_utils::increase_allowance(
+        &mut session,
+        ice_wood_pair.into(),
+        router.into(),
+        liquidity_minted,
+        BOB,
+    )
+    .unwrap();
+    let (_ice, _wood) = session
+        .execute(router.remove_liquidity(
+            ice.into(),
+            wood.into(),
+            liquidity_minted,
+            0,
+            0,
+            bob(),
+            deadline,
+        ))
+        .unwrap()
+        .result
+        .unwrap()
+        .unwrap();
+
+    // Fees now sent to `fee_to` address (CHARLIE).
+    let charlie_lp_balance = psp22_utils::balance_of(&mut session, ice_wood_pair.into(), charlie());
+    // Charlie withdraws his fees
+    psp22_utils::increase_allowance(
+        &mut session,
+        ice_wood_pair.into(),
+        router.into(),
+        charlie_lp_balance,
+        CHARLIE,
+    )
+    .unwrap();
+    let (protocol_fees_ice, protocol_fees_wood) = router::remove_liquidity(
+        &mut session,
+        router.into(),
+        ice.into(),
+        wood.into(),
+        charlie_lp_balance,
+        0,
+        0,
+        charlie(),
+    );
+    // We cannot assert exactly how much fees Charlie will get, due to roundings etc,
+    // but it should be close to the expected value
+    let received_protocol_fees = protocol_fees_ice + protocol_fees_wood;
+
+    // Trading fee is 0.3%.
```

Comment:
In this test it's 0.1%, right?

Reply:
And the comment [here](https://github.com/Cardinal-Cryptography/common-amm/pull/99/files#diff-2b32b45d84517f396b264a99964fafc11b56a2067ae311254c5109325118825fR462) should be updated.

Reply:
done